Board.py

```diff
@@ -1,16 +1,134 @@
+
+        source_column = next((col for col in self.columns if card in col), None)
+        if source_column:
+            cards_to_move = source_column[source_column.index(card):]
+
+            valid_sequence = all(
+                card.is_smaller_and_different_color(prev_card)
+                for card, prev_card in zip(cards_to_move[1:], cards_to_move)
+            )
+
+            if len(cards_to_move) <= self.empty_cells() and valid_sequence:
+                self.columns[next(i for i, col in enumerate(self.columns) if not col)].extend(cards_to_move)
+
+                index_of_card_to_move = source_column.index(card)
+                source_column[index_of_card_to_move:] = source_column[:index_of_card_to_move]
+
+                return True  # Move successful
+
+        return False  # Move unsuccessful
+
+    def __move_card_from_free_cell_to_card(self, card_to_move: Card, destination_card: Card) -> bool:
+        self.columns[next(i for i, col in enumerate(self.columns) if destination_card in col)].append(card_to_move)
+        self.free_cells[self.free_cells.index(card_to_move)] = None
+        return True
+
+
+
+    def __move_card_from_free_cell_to_empty_column(self, card_to_move: Card) -> bool:
+        self.columns[next(i for i, col in enumerate(self.columns) if not col)].append(card_to_move)
+        self.free_cells[self.free_cells.index(card_to_move)] = None
+        return True
+
+    def move_to_card(self, card_to_move: Card, destination_card: Card) -> bool:
+
+        if card_to_move.is_smaller_and_different_color(destination_card) and self.__is_on_top(destination_card):
+
+            # Check if card is in free cell
+            if card_to_move in self.free_cells:
+                return self.__move_card_from_free_cell_to_card(card_to_move, destination_card)
+
+            # Find the source column and destination column
+            dest_column = next((col for col in self.columns if destination_card in col), None)
+
+            source_column = next((col for col in self.columns if card_to_move in col), None)
+
+            # Check if both source and destination columns are found
+            if source_column and dest_column:
+                # Determine the cards to move
+                cards_to_move = source_column[source_column.index(card_to_move):]
+
+                valid_sequence = all(
+                    card.is_smaller_and_different_color(prev_card)
+                    for card, prev_card in zip(cards_to_move[1:], cards_to_move)
+                )
+
+                # Check if there are enough empty cells
+                if len(cards_to_move) <= self.empty_cells() and valid_sequence:
+                    # Move the cards
+                    dest_column.extend(cards_to_move)
+                    index_of_card_to_move = source_column.index(card_to_move)
+                    source_column[index_of_card_to_move:] = source_column[:index_of_card_to_move]
```

Comment:
Wydawało mi się, że jednak przesuwamy tylko jedną kartę, więc to jest trochę nadmiarowy kod, ale nadal będzie działał.

Reply:
Usuwam